translate.py

```diff
@@ -0,0 +1,90 @@
+from flask import abort
+import javascript.code_to_gast.js_main as js_main
+import python.code_to_gast.py_main as py_main
+import java.code_to_gast.java_main as java_main
+import shared.gast_to_code.gast_to_code_router as gtc
+import shared.gast_to_code.general_helpers as general_helpers
+from data_service import DataService
+import subprocess
+from shared.gast_to_code.converter_registry import ConverterRegistry
+
+
+def translate(input_code, input_lang, output_lang):
+    """
+    input_code: string representing input code
+    input_lang: string representing input language of code
+    output_lang: string representing output language of code
+    return: string representing output code or error message
+    """
+    error_handler = ConverterRegistry.get_converter(
+        output_lang).get_error_handler()
+
+    # code to gast
+    gast = code_to_gast_caller(input_code, input_lang, error_handler)
+    if (not is_valid_gast(gast)):
+        output_code = error_handler.compilation()
+        return output_code
+
+    #gast to code
+    output_code = gast_to_code_caller(gast, output_lang, error_handler)
+
+    # analytics
+    store_analytics_caller(input_code, output_code, input_lang, output_lang)
```

Comment:
I might be missing some context, but what is this doing during tests? All of these arguments are known by `translate`'s caller, I would prefer to have this stateful logging done there.

Reply:
@swalsh15 any insights? This was here before my PR